app/controllers/users/dossiers_controller.rb

```diff
@@ -671,5 +693,30 @@ def siret_params
     def commentaire_params
       params.require(:commentaire).permit(:body, piece_jointe: [])
     end
+
+    def redirect_if_hidden_or_deleted_dossier
+      dossier_id = params[:id]
+      hidden_dossier = hidden_dossier_for(dossier_id)
+
+      if hidden_dossier
+        return redirect_to corbeille_dossier_path(dossier_id)
+      end
+
+      deleted_dossier = deleted_dossier_for(dossier_id)
+
+      if deleted_dossier
+        return redirect_to supprime_dossier_path(dossier_id)
+      end
```

Comment:
on peut encore simplifier

```suggestion
      dossier_id = params[:id]

      if hidden_dossier_for(dossier_id[:id])
        return redirect_to corbeille_dossier_path(dossier_id)
      elsif deleted_dossier_for(dossier_id)
        return redirect_to supprime_dossier_path(dossier_id)
      end
```